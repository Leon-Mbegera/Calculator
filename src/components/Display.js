import React from 'react';
import PropTypes from 'prop-types';
import './calc.css';

function Display({ result }) {
  return <div className="display-screen">{result || '0'}</div>;
}

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
