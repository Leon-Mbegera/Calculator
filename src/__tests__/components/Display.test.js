import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Display from '../../components/Display';

describe('Display component', () => {
  render(<Display />);
  test('renders a default "0" on screen', () => {
    const value = screen.getByText('0');
    expect(value).toBeInTheDocument();
  });

  test('Display panel displays all calculator buttons', () => {
    const snap = renderer.create(<Display result="0" />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
