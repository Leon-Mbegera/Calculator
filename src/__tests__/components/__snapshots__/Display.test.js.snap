// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component Display panel displays all calculator buttons 1`] = `
<div
  className="display-screen"
>
  0
</div>
`;
